src/coreclr/jit/gentree.cpp

```diff
@@ -27525,11 +27525,11 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad(GenTree** pAddr) const
             case NI_Sve_GatherVectorUInt32WithByteOffsetsZeroExtend:
             case NI_Sve_GatherVectorUInt32ZeroExtend:
                 addr = Op(2);
-                if (varTypeIsSIMD(addr->gtType))
+                if (!varTypeIsI(addr))
                 {
-                    // The address is a vector of addresses.
-                    // Return true, but do not set pAddr.
-                    return true;
+                    // For some variants, the address is in vector.
+                    // Return false for such cases.
+                    return false;
```

Comment:
I was returning `true` here because I assumed we needed to track the fact there were addresses in `Op2`. Is it safe to not track that?

Reply:
Re-reading, the `OperIsMemoryLoad()` contract is to return true if this intrinsic may throw `NullReferenceException` of address is "null". Now, for certain non-faulting intrinsics, it should return `false`, but for `Gather*`, this should return `true`. 
Fixed it.

Reply:
> the OperIsMemoryLoad() contract is to return true if this intrinsic may throw NullReferenceException of address is "null".

That was the part I wasn't sure of.
Happy with the new version.